tests/test_migration.py

```
@@ -0,0 +1,169 @@
+        edge_cases = [
+            ("", ""),  # Empty URL
+            ("invalid-url", "invalid-url"),  # Invalid URL (should return as-is)
+            ("https://example.com", "https://example.com/"),  # No path
+            ("https://example.com/", "https://example.com/"),  # Root path
+        ]
+
+        for input_url, expected in edge_cases:
+            if expected:
+                result = normalize_url(input_url)
+                if input_url:  # Skip empty URL test as it may be handled differently
+                    assert result == expected or result == input_url, (
```

Ruff (0.12.2)

Use `result in (expected, input_url)` instead of multiple equality comparisons

Replace with `result in (expected, input_url)`

(SIM109)